src/gobprepare/prepare_client.py

```diff
@@ -183,27 +200,54 @@ def action_import_csv(self, action: dict):
             raise NotImplementedError
 
         rows_imported = importer.import_csv()
-        logger.info(f"Imported {rows_imported} rows from CSV in table {action['destination']}")
+        logger.info(f"Imported {rows_imported} rows from CSV to table {action['destination']}")
         return rows_imported
 
-    def _get_query(self, action: dict):
+    def action_import_api(self, action: dict):
+        """Import API action. Import API into destination database action.
+
+        :param action:
+        :return:
+        """
+        if self.destination['type'] == "postgres":
+            query = self._get_query(action)
+            importer = PostgresAPIImporter(self._dst_engine, action, query)
+        else:
+            raise NotImplementedError
+
+        rows_imported = importer.import_api()
+        logger.info(f"Imported {rows_imported} rows from API to table {action['destination']}")
+        return rows_imported
+
+    @staticmethod
+    def _read_query_string(query):
+        # Multiline queries are represented as lists in JSON. Join list as string
+        if isinstance(query, list):
+            return "\n".join(query)
+        return query
+
+    @staticmethod
+    def _read_file(file_name):
+        with open(file_name) as f:
+            return f.read()
+
+    def _get_query_per_source_type(self, action: dict, source_type, source_name):
         """Extracts query form action. Reads query from action or from file.
 
         :param action:
         :return:
         """
-        src = action.get('query_src')
+        src = action.get(source_type)
 
         if src == "string":
-            # Multiline queries are represented as lists in JSON. Join list as string
-            if isinstance(action["query"], list):
-                return "\n".join(action["query"])
-            return action["query"]
+            return self._read_query_string(action[source_name])
         elif src == "file":
-            with open(action["query"]) as f:
-                return f.read()
+            return self._read_file(action[source_name])
+
+        raise GOBException(f"Missing or invalid '{source_type}'")
 
-        raise GOBException("Missing or invalid 'query_src'")
+    def _get_query(self, action: dict):
+        return self._get_query_per_source_type(action, 'query_src', 'query')
```

Comment:
Why this extra complexity? _get_query_per_source_type is never used otherwise. Seems unnecessary and adds to the complexity in understanding the code.